pkg/tfbridge/auto_aliasing.go

```diff
@@ -125,409 +281,422 @@ func (info *ProviderInfo) ApplyAutoAliases() error {
 			}
 		}
 	}
+	return currentVersion, nil
+}
 
-	rMap := info.P.ResourcesMap()
-	dMap := info.P.DataSourcesMap()
+func (autoAliasHelper) computeMaxItemsOneOverrides(
+	schemaMap shim.SchemaMap,
+	schemaInfos map[string]*SchemaInfo,
+	history map[string]*fieldHistory,
+	over *maxItemsOneOverrides,
+) {
+	if schemaMap == nil {
+		return
+	}
+	fh := &fieldHistory{Fields: history}
+	walk.VisitSchemaMap(schemaMap, func(sp walk.SchemaPath, s shim.Schema) {
+		info := LookupSchemaInfoMapPath(sp, schemaInfos)
 
-	// Applying resource aliases adds new resources to providerInfo.Resources. To keep
-	// this process deterministic, we don't apply resource aliases until all resources
-	// have been examined.
-	//
-	// The same logic applies to datasources.
-	applyAliases := []func(){}
+		// If the user already set a MaxItemsOne override, respect it.
+		if info != nil && info.MaxItemsOne != nil {
+			return
+		}
 
-	for tfToken, computed := range info.Resources {
-		r, ok := rMap.GetOk(tfToken)
-		if !ok {
-			continue
+		actual := IsMaxItemsOne(s, info)
+		prev := fh.lookup(sp)
+
+		// Preserve historical MaxItemsOne if it is available and disagrees with current.
```

Comment:
Still thinking about rollout and correctness properties of this one..  It sounds like what we'd like to know from history, is the actual value (upstream+override) of MaxItemsOne, and that's it. It's currently has the *bool model but really we could be just storing the fields for which it's true, and assume it's false otherwise. 

That might not be the same semantic of what is stored there by the previous implementation. We'll need to check.. If that's the case we need some kind of migrate-on-read associated with this bridge change to roll out invisibly.